test/groovy/templates/PiperPipelineStageInitTest.groovy

```diff
@@ -128,6 +133,24 @@ class PiperPipelineStageInitTest extends BasePiperTest {
 
     }
 
+    @Test
+    void testCustomStashSettings() {
+        jryr.registerYaml('customStashSettings.yml',"unstash: source")
+        fileExistsRule.registerExistingFile('customStashSettings.yml')
+
+        jsr.step.piperPipelineStageInit(
+            script: nullScript,
+            juStabUtils: utils,
+            buildTool: 'maven',
+            customStashSettings: 'customStashSettings.yml',
+            stashSettings: 'com.sap.piper/pipeline/stashSettings.yml'
+        )
+
+        assertThat(stepsCalled, hasItems('checkout', 'setupCommonPipelineEnvironment', 'piperInitRunStageConfiguration', 'artifactPrepareVersion', 'pipelineStashFilesBeforeBuild'))
+        assertThat(stepsCalled, not(hasItems('slackSendNotification')))
+        assertThat(nullScript.commonPipelineEnvironment.configuration.stageStashes.unstash.toString(), is("source"))
```

Comment:
Could you also assert in the default case that [these stash settings](https://github.com/SAP/jenkins-library/blob/master/resources/com.sap.piper/pipeline/stashSettings.yml) are loaded? And in the ABAP case [these stash settings](https://github.com/SAP/jenkins-library/blob/master/resources/com.sap.piper/pipeline/abapEnvironmentPipelineStashSettings.yml)?
That would add the benefit of preventing somebody changing those stash settings undetected, as the unit tests would fail.